### PR TITLE
Add bindings for Safe.sol

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

Adds bindings for Safe.sol.

In order to get foundry to generate build artifacts I needed to import the Safe somewhere. 
I put them in Deploy.sol since that's where we know we'll need it for sure.

### CODE REVIEW

File: packages/contracts-bedrock/scripts/Deploy.s.sol [hunk 1]
```diff
@@ -9,6 +9,8 @@ import { stdJson } from "forge-std/StdJson.sol";
 import { Deployer } from "./Deployer.sol";
 import { DeployConfig } from "./DeployConfig.s.sol";
 
+import { Safe } from "safe-contracts/Safe.sol";
+import { SafeProxyFactory } from "safe-contracts/proxies/SafeProxyFactory.sol";
```
Comment (tynes) [inline packages/contracts-bedrock/scripts/Deploy.s.sol:13]: Any reason to have both but only build 1?

Comment (maurelian) [reply packages/contracts-bedrock/scripts/Deploy.s.sol:13]: Not really. I've added it.

**Review by tynes — APPROVED** [review]

File: .gitmodules [hunk 2]
```diff
@@ -13,3 +13,6 @@
 [submodule "packages/contracts-bedrock/lib/forge-std"]
 	path = packages/contracts-bedrock/lib/forge-std
 	url = https://github.com/foundry-rs/forge-std
+[submodule "packages/contracts-bedrock/lib/safe-contracts"]
```
Comment (tynes) [inline .gitmodules:16]: We should be sure that the correct tag is used, i dont think we want `master`

Comment (tynes) [reply .gitmodules:16]: https://github.com/safe-global/safe-contracts/tree/v1.4.0

`master` contains not prod ready code

Comment (maurelian) [reply .gitmodules:16]: great call out

**Comment by tynes** [comment]
```
Cloning into '/root/project/packages/contracts-bedrock/lib/safe-contracts'...
fatal: remote error: 
 safe-global/safe-contracts@v1.4.0 is not a valid repository name
Visit https://support.github.com/ for help
fatal: clone of 'https://github.com/safe-global/safe-contracts@v1.4.0' into submodule path '/root/project/packages/contracts-bedrock/lib/safe-contracts' failed
Failed to clone 'packages/contracts-bedrock/lib/safe-contracts' a second time, aborting
```

**Review by refcell — APPROVED** [review]

File: .gitmodules [hunk 3]
```diff
@@ -13,3 +13,10 @@
...
+	path = packages/contracts-bedrock/lib/safe-contracts
+	url = https://github.com/safe-global/safe-contracts
+  branch = v1.4.0
```
Comment (tynes) [inline .gitmodules]: looks like this was manually edited because of tabs vs spaces

Comment (tynes) [inline .gitmodules]: The commit `bf943f` corresponds to `v1.4.1` it seems, see the file below `lib/safe-contracts`

Comment (maurelian) [reply .gitmodules]: fixed.

Comment (maurelian) [reply .gitmodules]: I'm not sure how that commit is relevant. If you'd prefer 1.4.1 LMK and I'll bump this.

Comment (maurelian) [reply .gitmodules]: If that's the intent, please approve https://github.com/ethereum-optimism/optimism/pull/7299 and close this one. :)

Comment (maurelian) [reply .gitmodules]: OK, [the submodule](https://github.com/ethereum-optimism/optimism/pull/7293/files#diff-88425826d202bdcd3a5e4d48a8d23f29584966c42cbe5db3d29657368bbfa8c9) is now at the commit of [v1.4.0](https://github.com/safe-global/safe-contracts/releases/tag/v1.4.0) (`e870f514ad34cd9654c72174d6d4a839e3c6639f`).

**Pull request merged**